src/app/redux/history/reducer.js

```diff
@@ -0,0 +1,32 @@
+import {completeReducer, createReducer} from 'redux-recompose';
+import {actions} from './actions';
+import Immutable from 'seamless-immutable';
+
+const initialState = {
+  history: [],
+  historyFailure: null,
+  historyLoading: false,
+};
+const historyReducers = {
+  primaryActions: actions.primaryActions,
+  override: {
+    [actions.DEL_ALL_SUCCESS]: (state, action) =>
+      Immutable.merge(state, {history: action.payload}),
+    [actions.SET_HISTORY_SUCCESS]: (state, action) =>
+      Immutable.merge(state, {history: [...state.history, action.payload]}),
+    [actions.EDIT_EXPRESSION_HISTORY_SUCCESS]: (state, action) => {
+      const {newExpression, id} = action.payload;
+      let historyCopy = [...state.history];
+      historyCopy[id] = newExpression;
```

Comment:
Id is not the array position, id is a random generated string. This won't work when using the service.

Reply:
The service is added in the next PR, so change it in the other. In this he did not know how he was going to come from the service.